### ISSUE TITLE
Single quotes in task definition are not supported

### ISSUE THREAD

**Issue by lokst** [CircleCI-Public/aws-ecs-orb#18]
Refer to https://github.com/CircleCI-Public/aws-ecs-orb/issues/16 for an example of the error that will result when using the `update-task-definition` command when the existing task definition contains a single quote.

**Comment by chrisbloemker** [comment]
Is there a fix for this? I'm having trouble, maybe it's on my end.

The command in my `config.yml`

```
- aws-ecs/update-service:
          family: <<parameters.ecs-task>>
          cluster-name: <<parameters.ecs-cluster>>
          service-name: <<parameters.ecs-service>>
          container-image-name-updates: "container=<<parameters.ecs-task>>,tag=${CIRCLE_BRANCH}-${CIRCLE_SHA1}"
          verify-revision-is-deployed: true
```

Resulting in the following error:
```

REVISION=$(aws ecs register-task-definition \
    --family frontend \
    --container-definitions "${CCI_ORB_AWS_ECS_CONTAINER_DEFS}" \
    "$@" \
    --output text \
    --query 'taskDefinition.taskDefinitionArn')
echo "Registered task definition: ${REVISION}"
echo "export CCI_ORB_AWS_ECS_REGISTERED_TASK_DFN='${REVISION}'" >> $BASH_ENV
/tmp/.bash_env-5d2d32607067050008132abe-0-build: line 4: unexpected EOF while looking for matching `"'
/tmp/.bash_env-5d2d32607067050008132abe-0-build: line 10: syntax error: unexpected end of file

Error parsing parameter '--container-definitions': Expected: '=', received: 'EOF' for input:

^
Exited with code 255
```

My guess is it's failing on that last export to `$BASH_ENV` ?

**Comment by lokst** [comment]
Hi @chrisbloemker , yes, currently the `$BASH_ENV` cannot be sourced properly when there is a single quote in the task definition. Work on a fix for this is still in the backlog; is it feasible for you to check your task definition and remove single quote values found? Examples of where they could be found: passwords in connection URLs

**Comment by chrisbloemker** [comment]
@lokst Thank you for the quick reply, is this something I would be able to find in the ecs task definition json?

**Comment by chrisbloemker** [comment]
Oh now I can see that a randomly generated password had a single quote for the value from an environment variable, I'm guessing that's it.. Will fix and rerun the build :)

**Comment by Xheno** [comment]
I'd like to take this up as well, simply escaping the all single quotes (for bash) in the container definitions should do the trick.

**Comment by lokst** [comment]
@Xheno Thanks, that would be great! I'm not sure if it'll be tricky to escape the single quotes; it'd be awesome if the fix could be tested with a task definition containing a single quote!

**Comment by Xheno** [comment]
I'll test that it works before I make a PR.

**Comment by lokst** [comment]
Resolved by @Xheno in https://github.com/CircleCI-Public/aws-ecs-orb/pulls/67 ! The bug fix is available in `circleci/aws-ecs@1.0.1`  🎉
